mkosi/distributions/ubuntu.py

```diff
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+from typing import Set
+
+from ..backend import MkosiState, add_packages
+from .debian import DebianInstaller
+
+
+class UbuntuInstaller(DebianInstaller):
```

Comment:
Generally not a big fan of it either, but to put it in some words I heard recently: I think it can be done tastefully. :)

Without multiple inheritance this is a one way street of children overriding parents and I look at this more like drop in configuration and pushing this out to some other functions that we're then overwriting just reinvents inheritance again, so I'd like to keep it simple and stay with what's the language points one to.

Reply:
I'm not convinced. I think all the things we're overriding can easily be done locally in the ubuntu class before or after calling a shared install function. See my other comments

Reply:
I'm not convinced either. But maybe let's get this merged in this approximate form, and then maybe refactor later. How things are organized within the debian and ubuntu modules shouldn't impact the rest of the code, and we shouldn't get bogged down by this part.